src/main/java/org/cardanofoundation/cfexploreraggregator/addresstxcount/processor/AddressTxCountProcessor.java

```diff
@@ -102,9 +102,12 @@ public void handleCommitEvent(CommitEvent commitEvent) {
     }
 
     private void handleAddressAggregation(CommitEvent commitEvent, Map<String, Tuple<Long, Long>> snapshot) {
+        List<AddressTxCountEntity> allByAddressIn = addressTxCountRepository.findAllByAddressIn(snapshot.keySet().stream().toList());
+
         List<AddressTxCountEntity> entities = snapshot.entrySet().stream()
                 .map(entry -> {
-                    List<AddressTxCountEntity> byAddressOrderByTxCountDesc = addressTxCountRepository.findByAddressOrderBySlotDesc(entry.getKey());
+                    List<AddressTxCountEntity> byAddressOrderByTxCountDesc = allByAddressIn.stream().filter(addressTxCountEntity -> addressTxCountEntity.getAddress().equals(entry.getKey())).toList();
+
```

Comment:
* rename to something more meningful
* replace toList with findFirst or findAny no need to traverse the full list.

Reply:
I need the list, since I use it for deleting old entities